SpliceVault.pm

```diff
@@ -0,0 +1,212 @@
+ - Type: exon skipping (ES), cryptic donor (CD) or cryptic acceptor (CA)
+ - Transcript impact:
+   - For ES, describes skipped exons, e.g. ES:2 represents exon 2 skipping and
+     ES:2-3 represents skipping of exon 2 and 3
+   - For CD/CA, describes the distance from the annotated splice-site to the
+     cryptic splice-site with reference to the transcript (distances to negative
+     strand transcripts are reported according to the 5' to 3' distance)
+ - Percent of supporting samples: percent of samples supporting the event over
+   total samples where splicing occurs in that site (note this may be above 100%
+   if the event is seen in more samples than annotated splicing)
+ - Frameshift: inframe or out-of-frame event
+
+ The plugin also returns information specific to each splice site:
+ - Site position/type: genomic location and type (donor/acceptor) of the
+   splice-site predicted to be lost by SpliceAI. Cryptic positions are relative
+   to this genomic coordinate.
+ - Out of frame events: fraction of the top events that cause a frameshift. As
+   per https://pubmed.ncbi.nlm.nih.gov/36747048, sites with 3/4 or more in-frame
+   events are likely to be splice-rescued and not loss-of-function (LoF).
+ - Site sample count and max depth: sample count for this splice site and max
+   number of reads in any single sample representing annotated splicing in
+   Genotype-Tissue Expression (GTEx). This information allows to filter events
+   based on a minimum number of samples or minimum depth in GTEx.
+ - SpliceAI delta score (provided by SpliceVault)
+
+ Please cite the SpliceVault publication alongside the VEP if you use this
+ resource: https://pubmed.ncbi.nlm.nih.gov/36747048
+
+ The tabix utility must be installed in your path to use this plugin. The
+ SpliceVault TSV and respective index (TBI) for GRCh38 can be downloaded from
+ https://ftp.ensembl.org/pub/current_variation/SpliceVault/SpliceVault_data.tsv.gz
+
+ To filter results, please use filter_vep with the output file or standard
+ output. Documentation on filter_vep is available at:
+ https://www.ensembl.org/info/docs/tools/vep/script/vep_filter.html
+
+=cut
+
+package SpliceVault;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Utils::Sequence qw(reverse_comp);
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+  
+  # Check files in arguments
+  my @files; 
+  my $params = $self->params_to_hash();
+
+  for my $key (keys %{$params}) {
+    push @files, $params->{$key};
+  }
+
+  die "ERROR: add Tabix-indexed file, for example:\n" .
+    "  --plugin SpliceVault,file=/path/to/SpliceVault_data.tsv.gz \n"
```

Comment:
I would include 'GRCh38' in the file name for clarity

Reply:
Updated with:

```suggestion
    "  --plugin SpliceVault,file=/path/to/SpliceVault_data_GRCh38.tsv.gz \n"
```

Thanks for the suggestion.